cmd/allocator/main.go

```diff
@@ -179,10 +185,31 @@ func main() {
 	grpcServer := grpc.NewServer(opts...)
 	pb.RegisterAllocationServiceServer(grpcServer, h)
 
-	// serve GRPC for allocation
+	mux := gw_runtime.NewServeMux()
+	err = pb.RegisterAllocationServiceHandlerServer(context.Background(), mux, h)
+	if err != nil {
+		panic(err)
+	}
+
+	cfg := &tls.Config{}
+	if !h.tlsDisabled {
+		cfg.GetCertificate = h.getTLSCert
+	}
+	if !h.mTLSDisabled {
+		cfg.ClientAuth = tls.RequireAnyClientCert
+		cfg.VerifyPeerCertificate = h.verifyClientCertificate
+	}
+
+	// Create a Server instance to listen on port 8443 with the TLS config
+	server := &http.Server{
+		Addr:      ":8443",
+		TLSConfig: cfg,
+		Handler:   grpcHandlerFunc(grpcServer, mux),
+	}
+
 	go func() {
-		err := grpcServer.Serve(listener)
-		logger.WithError(err).Fatal("allocation service crashed")
+		err = server.ListenAndServeTLS("", "")
```

Comment:
Just to confirm - this is blank here, because the TLS config is handled above?

Reply:
Yeap, sicne we set custom `GetCertificate` function that gets used when the file names passed to this function are empty.
From the docs
```
// Filenames containing a certificate and matching private key for the
// server must be provided if neither the Server's TLSConfig.Certificates
// nor TLSConfig.GetCertificate are populated. 
```